dashboard/app/models/authentication_option.rb

```diff
@@ -70,4 +73,28 @@ def hash_email
     return unless email.present?
     self.hashed_email = AuthenticationOption.hash_email(email)
   end
+
+  private def email_must_be_unique
+    # skip the db lookup if possible
+    return unless email_changed?
```

Comment:
could condense to: `return unless email_changed? && email.present? && errors.blank?` (and same for lines 91-93 for hashed emails)